extensions/wikia/DesignSystem/scripts/DesignSystemGlobalNavigationOnSiteNotifications.js

```diff
@@ -0,0 +1,437 @@
+require(
+	['jquery', 'wikia.window', 'wikia.loader', 'wikia.mustache', 'wikia.log'],
+	function ($, window, loader, mustache, log) {
+		'use strict';
+
+		const notificationTypes = {
+			discussionUpvotePost: 'discussion-upvote-post',
+			discussionUpvoteReply: 'discussion-upvote-reply',
+			discussionReply: 'discussion-reply',
+			announcement: 'announcement'
+		}, logTag = 'on-site-notifications';
+
+		/**
+		 * Gets timestamp from ISO string date
+		 * @param {string} date
+		 * @returns {number} - timestamp
+		 */
+		function convertToTimestamp(date) {
+			return (new Date(date)).getTime() / 1000;
+		}
+
+		/**
+		 * Gets ISO string from timestamp
+		 * @param {string} timestamp
+		 * @returns {string} - ISO string
+		 */
+		function convertToIsoString(timestamp) {
+			return new Date(timestamp * 1000).toISOString();
+		}
+
+		function getSafely(obj, path) {
```

Comment:
This method probably should be extracted to a some object utils file. Additionally it should be unit tested.